nomad/container.py

```diff
@@ -165,18 +173,26 @@ def is_stopped(self):
         return self._container.status_code == constants.CONTAINER_STOPPED
 
     @property
+    def lxd_name(self):
+        """ Returns the name of the container that is used in the scope of LXD.
+
+        This name id supposed to be unique among all the containers managed by LXD.
+        """
+        # Note: all container names must be a valid hostname! That is: maximum 63 characters, no
+        # dots, no digit at first position, made entirely of letters/digits/hyphens, ...
+        if not hasattr(self, '_lxd_name'):
+            lxd_name_prefix = '{project_slug}-{name}'.format(
+                project_slug=slugify(self.project_name), name=self.name)
+            container_id = self.id
+            self._lxd_name = '{prefix}-{id}'.format(
+                prefix=lxd_name_prefix[:63 - len(container_id)], id=container_id)
```

Comment:
That's true, but the same could be said for Vagrant (which creates machine names using a similar pattern). My point is that it doesn't hurt to associate a unique identifier to every container that is created using lxd-nomad and to systematically use it to ensure that its LXD name is unique. Thus we can directly retrieve the containers associated with a specific project. The name are a bit longer, but the users should not use the "lxc" commands so often (they will always use the "local" names of the containers when running "nomad up/halt/...").